src/util/sock.cpp

```diff
@@ -106,6 +106,7 @@ bool Sock::Wait(std::chrono::milliseconds timeout, Event requested, Event* occur
     return true;
 #else
     if (!IsSelectableSocket(m_socket)) {
+        errno = WSAEINVAL;
```

Comment:
Isn't assigning to errno from non-libc code discouraged?

Reply:
I don't know, why would it be discouraged? There are already some assignments elsewhere in the code.

An alternative would be to introduce `Wait()`'s own error codes, all of which except one map 1:1 to errno values (as returned by `poll()` or `select()`) and one extra error code for this case where the socket is not selectable. Doesn't sound nice.

Reply:
I don't know. It really seems like something the operating system / C library does, not the user code.

Reply:
No errno, no problem.

(removed this change at the expense of a less detailed error message)